### PR TITLE
Product create

### PR DESCRIPTION
**Pull request by kalaivanidkpm**

### CODE REVIEW

File: Reactjs/src/components/product/ProductCreate.js [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+                break;
+        }
+        this.setState({formErrors:fieldValidationErrors,isValid:isValid },this.validForm);
+        console.log(this.state.formErrors); 
+      
+    }
+    validForm(){     
+      console.log(this.state)   ;
+        this.setState({
+          formValid:this.state.reed && this.state.pick
```
Comment (LogeshKandasamy) [inline Reactjs/src/components/product/ProductCreate.js:104]: warp and wept and width is not added in the formValid

File: Reactjs/src/components/product/ProductCreate.js [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+            case 'width':             
+                if(value.length <= 0){
+                    fieldValidationErrors.width = "Width is required. ";           
+                    isValid=false;
+                }else {isValid=true}                
+                break;          
+
+            case 'loomType':               
+                if(value.length <= 0){
+                    fieldValidationErrors.loomType = " LoomType is required. ";
```
Comment (LogeshKandasamy) [inline Reactjs/src/components/product/ProductCreate.js:89]: Can you move the string  " LoomType is required. "; to Constants.js

File: Reactjs/src/components/product/ProductCreate.js [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+          formValid:this.state.reed && this.state.pick
+        })          
+       
+    }
+
+    handleProductChange=(idp)=>(e)=>{
+        e.preventDefault();
+        const targetName = e.target.name;
+        const targetBValue = e.target.value;
+        console.log("Name : ",targetName);
```
Comment (LogeshKandasamy) [inline Reactjs/src/components/product/ProductCreate.js:113]: remove the console used for debg purpose

File: Reactjs/src/components/product/ProductCreate.js [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+        console.log(" Value ",targetBValue);
+        /* this.setState({[targetName]:targetBValue}, 
+            () => {this.validationField(targetName,targetBValue)
+            }); */
+        console.log(" state idp :", idp);
+        console.log(" state row :", this.state.rows);
+        const rows =[...this.state.rows]
+        console.log("rows ",rows);
+        
+        rows[idp]={[targetName]:targetBValue};
```
Comment (LogeshKandasamy) [inline Reactjs/src/components/product/ProductCreate.js:123]: targetBValue spell check

File: Reactjs/src/components/product/ProductCreate.js [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+        const rows =[...this.state.rows]
+        console.log("rows ",rows);
+        
+        rows[idp]={[targetName]:targetBValue}; 
+        //this.setState({rows});       
+        this.setState({[targetName]:targetBValue,rows} ,         
+        () => {       
+          {this.state.rows.map((item,i)=>{console.log(item)})}
+            this.validationField(targetName,targetBValue)
+            console.log(rows)
```
Comment (LogeshKandasamy) [inline Reactjs/src/components/product/ProductCreate.js:129]: remove the console

File: Reactjs/src/components/product/ProductCreate.js [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+        });         
+       
+    }
+
+    handleAddRow=()=>{
+        const tableItem={ reed:"",pick:"",warp:"",wept:"",width:""};
```
Comment (LogeshKandasamy) [inline Reactjs/src/components/product/ProductCreate.js:135]: loomtype is not added in the table row

File: Reactjs/src/components/product/ProductCreate.js [hunk 7]
```diff
@@ -0,0 +1,357 @@
...
+        this.setState({rows:[...this.state.rows,tableItem]})
+      
+      if (this.state.formErrors.reed.length != null &&
```
Comment (LogeshKandasamy) [inline Reactjs/src/components/product/ProductCreate.js:138]: warp, wept and width check is missing

File: Reactjs/src/components/product/ProductCreate.js [hunk 8]
```diff
@@ -0,0 +1,357 @@
...
+       /*   if (this.state.rows.length !=""){             
+          let obj={}
+          obj=this.state.rows          
+          this.setState({formValid:true,fomrData:obj})   */
+            
+            console.log("After CON check : ",this.state);
+            this.setState({subtitle:""})
+            this.setState({popupMsg:"Product Saved Successfully"})    
+            // pass data with axios post request              
+            const url= "https://vab0g55bof.execute-api.ap-southeast-1.amazonaws.com/nonprod/product";
```
Comment (LogeshKandasamy) [inline Reactjs/src/components/product/ProductCreate.js:172]: url should be  in the API Service.js

File: Reactjs/src/components/product/ProductCreate.js [hunk 9]
```diff
@@ -0,0 +1,357 @@
...
+            const productData =[{
+                                company:this.state.companyName,
+                                reed:this.state.reed, pick:this.state.pick,
+                                warp:this.state.warp, wept:this.state.wept,
+                                width:this.state.width, loomType:this.state.loomType,
+                                maxProductionPerDay:this.state.maxProduction,                                 
+                                }]          
+            const curHeaders={ 'Access-Control-Allow-Origin': '*','Content-Type': 'eg',}                   
+           //axios({method:'post',url:url,data:{productData},headers:curHeaders})
+            axios.post(url,{productData},{headers:curHeaders})
```
Comment (LogeshKandasamy) [inline Reactjs/src/components/product/ProductCreate.js:182]: send the post data to APIService

File: Reactjs/src/components/product/ProductCreate.js (hunk 4 above)
Comment (kalaivanidkpm) [reply Reactjs/src/components/product/ProductCreate.js:123]: Corrected.

File: Reactjs/src/components/product/ProductCreate.js (hunk 3 above)
Comment (kalaivanidkpm) [reply Reactjs/src/components/product/ProductCreate.js:113]: ok sir, Removed.

File: Reactjs/src/components/product/ProductCreate.js (hunk 1 above)
Comment (kalaivanidkpm) [reply Reactjs/src/components/product/ProductCreate.js:104]: warp, width,wept added in formvalid

File: Reactjs/src/components/product/ProductCreate.js (hunk 5 above)
Comment (kalaivanidkpm) [reply Reactjs/src/components/product/ProductCreate.js:129]: removed.

File: Reactjs/src/components/product/ProductCreate.js (hunk 6 above)
Comment (kalaivanidkpm) [reply Reactjs/src/components/product/ProductCreate.js:135]: loomtype added.

File: Reactjs/src/components/product/ProductCreate.js (hunk 7 above)
Comment (kalaivanidkpm) [reply Reactjs/src/components/product/ProductCreate.js:138]: checked warp wept and width.

File: Reactjs/src/components/product/ProductCreate.js (hunk 9 above)
Comment (kalaivanidkpm) [reply Reactjs/src/components/product/ProductCreate.js:182]: data send to the API services

File: Reactjs/src/components/product/ProductCreate.js (hunk 8 above)
Comment (kalaivanidkpm) [reply Reactjs/src/components/product/ProductCreate.js:172]: get the URL in API services

File: Reactjs/src/components/product/ProductCreate.js [hunk 10]
```diff
@@ -169,18 +172,19 @@ class ProductCreate extends React.Component{
...
                                 width:this.state.width, loomType:this.state.loomType,
                                 maxProductionPerDay:this.state.maxProduction,                                 
                                 }]          
-            const curHeaders={ 'Access-Control-Allow-Origin': '*','Content-Type': 'eg',}                   
+           // const curHeaders={ 'Access-Control-Allow-Origin': '*','Content-Type': 'eg',}                   
            //axios({method:'post',url:url,data:{productData},headers:curHeaders})
-            axios.post(url,{productData},{headers:curHeaders})
-               .then(response=>{                    
+            //axios.post(url,{productData},{headers:curHeaders})
+            APIService.getProductCreate({productData})
```
Comment (LogeshKandasamy) [inline Reactjs/src/components/product/ProductCreate.js:186]: get is used for read the data frm backend , 

Correct the method name

File: Reactjs/src/service/APIService.js [hunk 11]
```diff
@@ -2,14 +2,16 @@ import axios from 'axios'
 
 
 const CUST_REST_API_URL = 'https://vab0g55bof.execute-api.ap-southeast-1.amazonaws.com/nonprod/customer';
+const PRODUCT_REST_API_URL = 'https://vab0g55bof.execute-api.ap-southeast-1.amazonaws.com/nonprod/product';
```
Comment (LogeshKandasamy) [inline Reactjs/src/service/APIService.js:5]: Customer and product url is almost same 

last portion varies /customer , / product

modify more simpler way

File: Reactjs/src/service/APIService.js [hunk 12]
```diff
@@ -2,14 +2,16 @@ import axios from 'axios'
...
 
 const UPLOAD_HEADER= {
     headers: {'Access-Control-Allow-Origin': '*',
     'Content-Type': 'multipart/form-data'
     }
 };
 const HEADER= {
-    headers: {'Access-Control-Allow-Origin': '*'
+    headers: {'Access-Control-Allow-Origin': '*',
+    'Content-Type': 'eg'
```
Comment (LogeshKandasamy) [inline Reactjs/src/service/APIService.js:14]: Content-Type is 'eg' 

invalid content type

**Review by LogeshKandasamy — CHANGES_REQUESTED** [review]

File: Reactjs/src/service/APIService.js (hunk 12 above)
Comment (kalaivanidkpm) [reply Reactjs/src/service/APIService.js:14]: Removed

File: Reactjs/src/service/APIService.js (hunk 11 above)
Comment (kalaivanidkpm) [reply Reactjs/src/service/APIService.js:5]: URl modified and checked  for both Customer and Product.

File: Reactjs/src/components/product/ProductCreate.js (hunk 10 above)
Comment (kalaivanidkpm) [reply Reactjs/src/components/product/ProductCreate.js:186]: name changed.

File: Reactjs/src/components/product/ProductCreate.js (hunk 2 above)
Comment (kalaivanidkpm) [reply Reactjs/src/components/product/ProductCreate.js]: MODIFIED

**Review by LogeshKandasamy — APPROVED** [review]

**Pull request merged**